src/client.coffee

```diff
@@ -34,34 +35,55 @@ module.exports = class Client
   constructor: (provider_key, options) ->
```

Comment:
Try something like this:

```coffee
class Client
  # Handles old syntax
  #   client = new Client(provider_key, {host: 'gw.example.com'})
  #
  # OR
  #
  # New syntax does not handle provider_key
  #   client = new Client({service_token: 'TOKEN', host: 'gw.example.com'})
  #
  # This will throw error:
  #   client = new Client({provider_key: 'PROVIDER_KEY'})
  constructor: (provider_key, options) ->
    if provider_key instanceof Object
      opts = provider_key
    else
      opts = options?
      @provider_key = provider_key

    @service_token = opts?.service_token
    @host = opts?.host ? 'su1.3scale.net'

    unless (@service_token? or @provider_key?)
      throw 'No provider key or service token provided'
```